cpp/include/doxygen_groups.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file
+ * @brief Doxygen group definitions
+ */
+
+// This header is only processed by doxygen and does
+// not need to be included in any source file.
+// Below are the main groups that doxygen uses to build
+// the Modules page in the specified order.
+//
+// To add a new API to an existing group, just use the
+// @ingroup tag in the API's doxygen comment.
+// Add a new group by first specifying in the hierarchy below.
+
+/**
+ * @defgroup cuspatial_constants Constants
+ * @{
+ *   @brief Constants used in cuspatial APIs
+ *   @file constants.hpp
+ * @}
+ * @defgroup spatial_api Spatial APIs
+ * @{
+ *      @defgroup coordinate_transform Coordinate Transform
+ *      @{
+ *          @brief APIs to transform coordinates
+ *
+ *          This module contains APIs that transforms cartesian and geodesic coordinates.
+ *          @file coordinate_transform.hpp
+ *      @}
+ *      @defgroup distance Distance
+ *      @{
+ *          @brief Distance computation APIs
+ *
+ *          @file linestring_distance.hpp
+ *          @file hausdorff.hpp
+ *          @file haversine.hpp
+ *          @file haversine.cuh
+ *      @}
+ *      @defgroup spatial_relationship Spatial Relationship
+ *      @{
+ *          @brief APIs related to spatial relationship
+ *
+ *          @file point_in_polygon.hpp
+ *          @file polygon_bounding_box.hpp
+ *          @file polyline_bounding_box.hpp
+ *          @file spatial_window.hpp
+ *      @}
+ *      @defgroup cubic_spline Cubic Spline
+ *      @{
+ *          @brief APIs related to cubic splines
+ *          @file cubic_spline.hpp
+ *      @}
+ * @}
+ * @defgroup trajectory_api Trajectory APIs
+ * @{
+ *      @brief APIs related to trajectories
+ *      @file trajectory.hpp
+ * @}
+ * @defgroup spatial_indexing Spatial Indexing
+ * @{
+ *      @brief APIs related to spatial indexing
+ *      @file point_quadtree.hpp
+ * @}
+ * @defgroup spatial_join Spatial Join
+ * @{
+ *      @brief APIs related to spatial join
+ *      @file spatial_join.hpp
+ * @}
+ * @defgroup cuspatial_types Types
+ * @{
+ *      @defgroup type_definition_operator Definition and Operators
```

Comment:
I don't think this group is necessary. The types can live in the top level of the Types group. The factories can be in a subgroup.